pkg/pod/pods.go

```diff
@@ -562,72 +556,77 @@ func (p *Pod) refreshState() error {
 	p.isDeleting = false
 	p.isGone = false
 
-	// we try open the pod in all possible directories, in the same order the states occur
-	_, err := os.Stat(p.embryoPath())
-	if err == nil {
-		p.isEmbryo = true
-	} else if os.IsNotExist(err) {
-		_, err := os.Stat(p.preparePath())
+	// dirStates is a list of directories -> state that directory existing
+	// implies.
+	// Its order matches the order states occur.
+	dirStates := []struct {
```

Comment:
I gave it a try, but I'm not sure it actually ends up being that much more readable with only a trivial mount more work. [here](https://github.com/euank/rkt/commit/d9ba603974498ebbfbb9a9a1e906d0c80fbf96d7)'s the wip exploration, but the end result neither worked nor looked *that* much prettier.

Let's just improve it incremental so long as we all agree this specific change is good.